### PR TITLE
add sharc dataset

### PR DESCRIPTION
**Pull request by patil-suraj**
This PR adds the ShARC dataset.

More info:
https://sharc-data.github.io/index.html

### CODE REVIEW

File: datasets/sharc/sharc.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                    "scenario": datasets.Value("string"),
+                    "history": [
+                        {"follow_up_question": datasets.Value("string"), "follow_up_answer": datasets.Value("string")}
+                    ],
+                    "evidence": [
+                        {"follow_up_question": datasets.Value("string"), "follow_up_answer": datasets.Value("string")}
+                    ],
+                    "answer": datasets.Value("string"),
+                    "negative_question": datasets.Value("bool_"),
+                    "negative_scenario": datasets.Value("bool_"),
```
Comment (patil-suraj) [inline datasets/sharc/sharc.py:73]: The dataset also provides ids of utterances where question or scenario is negative, so added two extra keys to indicate `negative_question` and `negative_scenario`

File: datasets/sharc/sharc.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                }
+            ),
+            supervised_keys=None,
+            homepage="https://github.com/Websail-NU/CODAH",
```
Comment (lhoestq) [inline datasets/sharc/sharc.py:77]: could you update the homepage ?

File: datasets/sharc/dataset_infos.json [hunk 3]
```diff
@@ -0,0 +1 @@
...stion": {"dtype": "string", "id": null, "_type": "Value"}, "follow_up_answer": {"dtype": "string", "id": null, "_type": "Value"}}], "evidence": [{"follow_up_question": {"dtype": "string", "id": null, "_type": "Value"}, "follow_up_answer": {"dtype": "string", "id": null, "_type": "Value"}}], "answer": {"dtype": "string", "id": null, "_type": "Value"}, "negative_question": {"dtype": "bool_", "id": null, "_type": "Value"}, "negative_scenario": {"dtype": "bool_", "id": null, "_type": "Value"}}, "post_processed": null, "supervised_keys": null, "builder_name": "sharc", "config_name": "sharc", "version": {"version_str": "1.0.0", "description": null, "major": 1, "minor": 0, "patch": 0}, "splits": {"train": {"name": "train", "num_bytes": 15088577, "num_examples": 21890, "dataset_name": "sharc"}, "validation": {"name": "validation", "num_bytes": 1469172, "num_examples": 2270, "dataset_name": "sharc"}}, "download_checksums": {"https://sharc-data.github.io/data/sharc1-official.zip": {"num_bytes": 5230207, "checksum": "0c185809b807c00df05eeca2c504dc5d6d0ecbf98cf60ea51f7d1e690493b69a"}}, "download_size": 5230207, "post_processing_size": null, "dataset_size": 16557749, "size_in_bytes": 21787956}}
```
Comment (lhoestq) [inline datasets/sharc/dataset_infos.json:1]:
```suggestion
{"sharc": {"description": "ShARC is a Conversational Question Answering dataset focussing on question answering from texts containing rules. The goal is to answer questions by possibly asking follow-up questions first. It is assumed assume that the question is often underspecified, in the sense that the question does not provide enough information to be answered directly. However, an agent can use the supporting rule text to infer what needs to be asked in order to determine the final answer.\n", "citation": "@misc{saeidi2018interpretation,\n      title={Interpretation of Natural Language Rules in Conversational Machine Reading},\n      author={Marzieh Saeidi and Max Bartolo and Patrick Lewis and Sameer Singh and Tim Rockt\u00e4schel and Mike Sheldon and Guillaume Bouchard and Sebastian Riedel},\n      year={2018},\n      eprint={1809.01494},\n      archivePrefix={arXiv},\n      primaryClass={cs.CL}\n}\n", "homepage": "https://sharc-data.github.io/index.html", "license": "", "features": {"id": {"dtype": "string", "id": null, "_type": "Value"}, "utterance_id": {"dtype": "string", "id": null, "_type": "Value"}, "source_url": {"dtype": "string", "id": null, "_type": "Value"}, "snippet": {"dtype": "string", "id": null, "_type": "Value"}, "question": {"dtype": "string", "id": null, "_type": "Value"}, "scenario": {"dtype": "string", "id": null, "_type": "Value"}, "history": [{"follow_up_question": {"dtype": "string", "id": null, "_type": "Value"}, "follow_up_answer": {"dtype": "string", "id": null, "_type": "Value"}}], "evidence": [{"follow_up_question": {"dtype": "string", "id": null, "_type": "Value"}, "follow_up_answer": {"dtype": "string", "id": null, "_type": "Value"}}], "answer": {"dtype": "string", "id": null, "_type": "Value"}, "negative_question": {"dtype": "bool_", "id": null, "_type": "Value"}, "negative_scenario": {"dtype": "bool_", "id": null, "_type": "Value"}}, "post_processed": null, "supervised_keys": null, "builder_name": "sharc", "config_name": "sharc", "version": {"version_str": "1.0.0", "description": null, "major": 1, "minor": 0, "patch": 0}, "splits": {"train": {"name": "train", "num_bytes": 15088577, "num_examples": 21890, "dataset_name": "sharc"}, "validation": {"name": "validation", "num_bytes": 1469172, "num_examples": 2270, "dataset_name": "sharc"}}, "download_checksums": {"https://sharc-data.github.io/data/sharc1-official.zip": {"num_bytes": 5230207, "checksum": "0c185809b807c00df05eeca2c504dc5d6d0ecbf98cf60ea51f7d1e690493b69a"}}, "download_size": 5230207, "post_processing_size": null, "dataset_size": 16557749, "size_in_bytes": 21787956}}
```

update the homepage

File: datasets/sharc/sharc.py (hunk 2 above)
Comment (lhoestq) [inline datasets/sharc/sharc.py:77]:
```suggestion
            homepage="https://sharc-data.github.io/index.html",
```

update the homepage

**Review by lhoestq — APPROVED** [review]
Nice thank you for adding this one.

File: datasets/sharc/sharc.py (hunk 2 above)
Comment (lhoestq) [reply datasets/sharc/sharc.py]: actually I just did it using the webpage mentioned in the dataset card

Comment (patil-suraj) [reply datasets/sharc/sharc.py]: Thanks Quentin!

**Pull request merged**